### DISCUSSION TITLE
Search results limited to specific collection with REGEX – best practice advise or a bug in 3.2.27

### DISCUSSION THREAD

**Discussion by janpersiel** [statamic/cms#5009] [Q&A] [answered]
Hi there,

I just updated to 3.2.28 and found my custom search breaking. It worked all the way up to 3.2.6 It should look like this: https://gingingin.de/suche?q=london

In my search results I list various clusters of content to make it easier to find what you need. Since the problem seems to be the same for all sections I will list only one code snippet which seems to be troublesome.

<img width="949" alt="image" src="https://user-images.githubusercontent.com/6762951/147981553-34191059-2220-4418-956a-0ff9c18d218f.png">

`{{ search:results collection:regex="/gin/" as="spirits" sort="date:desc" limit="6" index="default" }}
    {{ if no_results }}
    {{ else }}
    <section class="cathero latest col-span-12">
      <div class="regal col-span-12 sm:col-start-2 sm:col-span-10 md:col-start-3 md:col-span-8">
        {{ spirits }}
        <a href="{{ url }}" title="Mehr {{ spirits_producer_name }} {{ title }} Details">
        <div class="flasche">
          {{ spirit_bottle_image }}
            <div class="schatten"></div>
            <div class="bild">
              {{ responsive:url ratio="0.5"}}
            </div>
          {{ /spirit_bottle_image }}
        </div>
        </a>
        {{ /spirits }}
      </div>
    </section>
    {{ /if }}
    {{ /search:results }}`

This breaks and I could not find any other documentation on whether this is still a valid way of getting to what I am trying to achieve.

The error thrown looks like this:

> Error
> Object of class Statamic\Entries\Entry could not be converted to string

<img width="363" alt="image" src="https://user-images.githubusercontent.com/6762951/147981217-affe0ab2-13d2-4caf-b199-867f9ce1efb5.png">

So I am wondering whether this is a bug or just a wrong way of doing it despite having created it originally on this post/answer https://statamic.com/forum/4448-search-results-from-multiple-collections-paramater-syntax …

Thanks for the help / advise 🙏

**Comment by jasonvarga** [comment, marked as answer]
If it used to work then I'd say it's a bug. I don't think there's anything wrong with what you're trying to do.

**Reply by janpersiel** [reply]
Okay, I'll file a bug. It did work since I created this in August and for several updates.

**Reply by ryanmitchell** [reply]
Can you try updating to 3.2.29 as this error is the same as what [this commit](https://github.com/statamic/cms/commit/34550e17ad20eb7893f21cf26509d7c303ac595b) fixes

**Reply by janpersiel** [reply]
> Can you try updating to 3.2.29 as this error is the same as what [this commit](https://github.com/statamic/cms/commit/34550e17ad20eb7893f21cf26509d7c303ac595b) fixes

Sorry @ryanmitchell I did and closed the issue I created earlier. Simply forgot to post it here as well. It does work indeed in 3.2.29
